### PR TITLE
fix(deps): upgrade some dependencies to resolve CVE-2020-29652

### LINKED ISSUE #68: trivy security scan trips on CVE-2020-29652
Our CI pipeline runs trivy on our docker images, today it started to throw an error on version 0.4.1 of grpc_health_probe:
```
usr/local/bin/grpc_health_probe
===============================
Total: 1 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 1, CRITICAL: 0)
+---------------------+------------------+----------+------------------------------------+------------------------------------+---------------------------------------+
|       LIBRARY       | VULNERABILITY ID | SEVERITY |         INSTALLED VERSION          |           FIXED VERSION            |                 TITLE                 |
+---------------------+------------------+----------+------------------------------------+------------------------------------+---------------------------------------+
| golang.org/x/crypto | CVE-2020-29652   | HIGH     | v0.0.0-20200622213623-75b288015ac9 | v0.0.0-20201216223049-8b5274cf687f | golang: crypto/ssh: crafted           |
|                     |                  |          |                                    |                                    | authentication request can            |
|                     |                  |          |                                    |                                    | lead to nil pointer dereference       |
|                     |                  |          |                                    |                                    | -->avd.aquasec.com/nvd/cve-2020-29652 |
+---------------------+------------------+----------+------------------------------------+------------------------------------+---------------------------------------+
```

Not sure what the impact for other people might be. Also tricky as this is basically coming from another package that (inadvertently) is used.

### PR DESCRIPTION
**Pull request by rio**
fixes #68

It is classified as severity HIGH and even though grpc-health-probe doesn't really
use that code path resolving this is important because needing people to build
in exceptions to CVEs in their pipelines is a brittle process to say the least.

### CODE REVIEW

**Comment by rio** [comment]
@googlebot I signed it!

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,6 @@ out/
 
 # Output of the go coverage tool, specifically when used with LiteIDE
 *.out
+
+# ignore any local vendored packages
+vendor
```
Comment (ahmetb) [inline .gitignore:20]: any particular reason to update this file?

Comment (rio) [reply .gitignore:20]: Not everybody uses the `GOPATH` but instead simply runs `go mod vendor` to pull in any project dependencies.  This line prevents any accidental commits of all vendored packages by overzealous `git commit -a` commands. Just a little safety precaution ;)

**Comment by ahmetb** [comment]
Feel free to do `go get -u ./...` while we're at it.

File: .gitignore (hunk 1 above)
Comment (ahmetb) [reply .gitignore:20]: I'm not sure if that's entirely correct. Yeah, nobody uses GOPATH anymore but `go mod vendor` is not how you pull in deps to build a program.

This is not a library repository (it is a program) and in the future we can choose to actually vendor in the deps. Please undo if it's not helping this PR.

Comment (rio) [reply .gitignore:20]: >  go mod vendor is not how you pull in deps to build a program.

This I don't necessarily agree with but

> Please undo if it's not helping this PR.

Is quite valid :)

**Comment by ahmetb** [comment]
Thank you. 🙏

**Comment by rio** [comment]
no, thank you :bow:

**Pull request merged**